### DISCUSSION TITLE
(suspected false positive) Vulnerabilities in datalust/seq:2023.1.9133

### DISCUSSION THREAD

**Discussion by ThomasErvik** [datalust/seq-tickets#1820] [General]
Hi,

Is it possible to patch these issues in the next release?
![image](https://user-images.githubusercontent.com/88371500/233362444-549f42ca-ed2a-4e74-9fbc-90199188f1db.png)

- Thomas

**Comment by nblumhardt** [comment]
Thanks for reaching out, Thomas! We can definitely update anything that's outdated in our next patch, due shortly.

The files in question are all bundled at their 7.x versions, however. Seq 2023.1 is built with the .NET 7 SDK and notwithstanding some issue we're yet to find, uses a framework version much more recent than the versions mentioned in that screenshot.

Would it be possible to reach out to us via `support@datalust.co` to share more details of the scan and narrow this down a bit? Thanks!

**Reply by nblumhardt** [reply]
This looks like a case of https://github.com/dotnet/sdk/issues/30659 - the scanner is attempting to use the .deps.json file included in the package, without checking the .dll versions that dependencies will resolve to at runtime.

The `datalust/seq` image contains only the .NET 7 SDK versions of these DLLs. Hopefully either the scanner will get smarter, or some upstream solution like https://github.com/dotnet/linker/pull/3193 will change the layout of the .deps.json  file in a way that the scanner will accept or ignore. We'll give that a try in the upcoming patch.

Having trawled through the container filesystem by hand just now, it's fairly safe to say this one's a false positive.

**Reply by ThomasErvik** [reply]
Thanks for the info. 
I'll do some more checking on our end. 
We had this System.Drawing issue in some our projects also.

**Reply by todthomson** [reply]
Hi @ThomasErvik,

FYI, we upgraded our direct dependencies to the latest, and now when I run `dotnet list package --vulnerable` against the source code of Seq version `2023.1.9229` (the latest as of April 24th 2023) it reports no vulnerabilities for any packages that the Seq solution is referring to directly. 

```
The given project `?` has no vulnerable packages given the current sources.
```

We haven't promoted this to the `latest` tag yet in Docker Hub, but we will do at some stage in the coming days (presuming there are no further issues reported).

Please feel free have a go with Seq `2023.1.9229` and let us know if that improves things for you (or not). 😄 

Docker: https://hub.docker.com/layers/datalust/seq/2023.1.9229/images/sha256-aac844ce1c6c3c449f2bbbd07ac0cbd827ec4956cf10b82e818396efecd5eead?context=explore.

Download: https://datalust.co/download.

Cheers,

Tod.

**Comment by ThomasErvik** [comment]
Hi,

Thanks.
I’ll have a look at it.

-Thomas

________________________________
